### PR TITLE
Usethe directionality of dative bonds in substructure matching

### PR DESCRIPTION
**Pull request by greglandrum**
This is a simple patch to change the behavior so that `[Fe]->[Cu]` no longer matches `[Fe]<-[Cu]`.

### CODE REVIEW

File: Code/GraphMol/Substruct/SubstructUtils.cpp [hunk 1]
```diff
@@ -24,8 +24,8 @@ bool atomCompat(const ATOM_SPTR &a1, const ATOM_SPTR &a2,
   // " " << a2->getIdx() << std::endl;
   bool res;
   if (useQueryQueryMatches && a1->hasQuery() && a2->hasQuery()) {
-    res = static_cast<QueryAtom *>(a1.get())
-              ->QueryMatch(static_cast<QueryAtom *>(a2.get()));
+    res = static_cast<QueryAtom *>(a1.get())->QueryMatch(
```
Comment (bp-kelley) [inline Code/GraphMol/Substruct/SubstructUtils.cpp]: Is this an unrelated bug fix?

File: Code/GraphMol/Substruct/SubstructUtils.cpp [hunk 2]
```diff
@@ -55,11 +55,19 @@ bool bondCompat(const BOND_SPTR &b1, const BOND_SPTR &b2,
...
   bool res;
   if (useQueryQueryMatches && b1->hasQuery() && b2->hasQuery()) {
-    res = static_cast<QueryBond *>(b1.get())
-              ->QueryMatch(static_cast<QueryBond *>(b2.get()));
+    res = static_cast<QueryBond *>(b1.get())->QueryMatch(
+        static_cast<QueryBond *>(b2.get()));
   } else {
     res = b1->Match(b2);
   }
+  if (res && b1->getBondType() == Bond::DATIVE &&
```
Comment (bp-kelley) [inline Code/GraphMol/Substruct/SubstructUtils.cpp]: What happens when only one ( or the query ) is dative?  Is this a match or not a match?

File: Code/GraphMol/Substruct/SubstructUtils.cpp (hunk 1 above)
Comment (greglandrum) [inline Code/GraphMol/Substruct/SubstructUtils.cpp]: it's just a formatting change made by clang-format

File: Code/GraphMol/Substruct/SubstructUtils.cpp (hunk 2 above)
Comment (greglandrum) [inline Code/GraphMol/Substruct/SubstructUtils.cpp]: Generally mismatched bond types lead to the bonds not matching, so res would already be false and we wouldn't get here.
In the event that the query is generic enough to match the dative bond without being a dative bond itself, I think it's probably safest to assume a match without checking the direction.

**Comment by greglandrum** [comment]
@bp-kelley : I'm pretty sure we agreed that this one is ok, but I want to confirm before I merge it.
